format/caff/caff.go

```diff
@@ -0,0 +1,205 @@
+package caff
+
+import (
+	"bytes"
+	"compress/flate"
+	"io"
+
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/pkg/bitio"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/interp"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+var probeGroup decode.Group
+
+func init() {
+	interp.RegisterFormat(
+		format.CAFF,
+		&decode.Format{
+			Description: "Live2D Cubism archive",
+			Groups:      []*decode.Group{format.Probe},
+			DecodeFn:    decodeCAFF,
+			DefaultInArg: format.CAFF_In{
+				Uncompress: true,
+			},
+			Dependencies: []decode.Dependency{
+				{Groups: []*decode.Group{format.Probe}, Out: &probeGroup},
+			},
+		})
+}
+
+const (
+	imageFormatUnknown   = 0x00
+	imageFormatPNG       = 0x01
+	imageFormatNoPreview = 0x7F
+)
+
+const (
+	colorTypeUnknown   = 0x00
+	colorTypeARGB      = 0x01
+	colorTypeRGB       = 0x02
+	colorTypeNoPreview = 0x7F
+)
+
+const (
+	compressOptionRaw   = 0x10
+	compressOptionFast  = 0x21
+	compressOptionSmall = 0x25
+)
+
+var imageFormatNames = scalar.UintMapSymStr{
+	imageFormatUnknown:   "unknown",
+	imageFormatPNG:       "png",
+	imageFormatNoPreview: "no_preview",
+}
+
+var colorTypeNames = scalar.UintMapSymStr{
+	colorTypeUnknown:   "unknown",
+	colorTypeARGB:      "argb",
+	colorTypeRGB:       "rgb",
+	colorTypeNoPreview: "no_preview",
+}
+
+var compressOptionNames = scalar.UintMapSymStr{
+	compressOptionRaw:   "raw",
+	compressOptionFast:  "fast",
+	compressOptionSmall: "small",
+}
+
+type fileInfoListEntry struct {
+	filePath       string
+	startPos       int64
+	fileSize       int
+	isObfuscated   bool
+	compressOption uint8
+}
+
+func decodeVersion(d *decode.D) {
+	d.FieldU8("major")
+	d.FieldU8("minor")
+	d.FieldU8("patch")
+}
+
+func decodeCAFF(d *decode.D) any {
+	var ci format.CAFF_In
+	d.ArgAs(&ci)
+
+	var obfsKey uint64
+
+	obfsU8 := func(d *decode.D) uint64 { return d.U8() ^ (obfsKey & 0xFF) }
+	obfsU32 := func(d *decode.D) uint64 { return d.U32() ^ (obfsKey & 0xFFFFFFFF) }
```

Comment:
No a big thing but rest of the code uses lowercase and grouping for hex, binary literals

Reply:
Updated for consistency 👍